vars/infra.groovy

```diff
@@ -18,3 +18,63 @@ Object withDockerCredentials(Closure body) {
         echo 'Cannot use Docker credentials outside of the trusted environment'
     }
 }
+
+void stashJenkinsWar(jenkins, stashName = "jenkinsWar") {
+    def isVersionNumber = (jenkins =~ /^\d+([.]\d+)*$/).matches()
+    def isLocalJenkins = jenkins.startsWith("file://")
+    def mirror = "http://mirrors.jenkins.io/"
+
+    def jenkinsURL
+
+    if (jenkins == "latest") {
+        jenkinsURL = mirror + "war/latest/jenkins.war"
+    } else if (jenkins == "latest-rc") {
+        jenkinsURL = mirror + "/war-rc/latest/jenkins.war"
+    } else if (jenkins == "lts") {
+        jenkinsURL = mirror + "war-stable/latest/jenkins.war"
+    } else if (jenkins == "lts-rc") {
+        jenkinsURL = mirror + "war-stable-rc/latest/jenkins.war"
+    }
+
+    if (isLocalJenkins) {
+        if (!fileExists(jenkins - "file://")) {
+            error "Specified Jenkins file does not exists"
+        }
+    }
+    if (!isVersionNumber && !isLocalJenkins) {
+        echo 'Checking whether Jenkins WAR is available…'
+        sh "curl -ILf ${jenkinsURL}"
+    }
+
+    List<String> toolsEnv = [
```

Comment:
I have generalized it in #21, will decouple it to a separate PR

Reply:
@oleg-nenashev Great, I will incorporate that as soon as the PR exists, please note that this PR will also generalize and consolidate code at that at this moment is common between runATH and runPCT like downloading and stashing the jenkins war and the ensureInNode method.

Reply:
Created https://github.com/jenkins-infra/pipeline-library/pull/36